### PR TITLE
Add Contribution Readme

### PR DESCRIPTION
**Pull request by umfranzw**
Add a CONTRIBUTING.md file that contains information users will need if they want to contribute to the repository.

This is based on the template located here:
https://github.com/ROCm/.github/blob/main/docs/templates/contributing_template.md

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+The following subdirectories are organized according to software-scope level:
+* grid/ contains headers for grid-level operations (barriers, queues, etc.)
+* thread/ contains headers for thread-level operations (load/store, scan, reduce, etc.)
+
+Finally, the iterator/ subdirectory provides access to the iterators that are used to interact with most algorithms in the library.
+
+Back at the root level, you can find tests and benchmarks located inside directories of the same name. 
+There is also an examples folder that contains a number of sample API use cases.
+
+## Coding Style ##
```
Comment (umfranzw) [inline CONTRIBUTING.md:66]: This section is copied from rocSPARSE's CONTRIBUTING.md. I believe most of this holds true for us here in hipCUB as well, but I'm not 100% certain. If you see anything you think should be modified, feel free to let me know.

Comment (randyh62) [reply CONTRIBUTING.md:66]: I made minor edits to your text.

**Review by randyh62 — APPROVED** [review]
Looks good to me.

**Review by RobsonRLemos — APPROVED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+When a pull request is created, a number of automated checks are run. These checks:
+- test the change on various OS platforms (Ubuntu, RHEL, etc.)
+- run on different GPU architectures (MI-series, Radeon series cards, etc.)
+- run benchmarks to check for performance degredation
+
+In order for change to be accepted:
+- it must pass all of the automated checks
+- it must undergo a code review
+
+The GitHub "Issues" tab may also used to discuss ideas surrounding particular features or changes before raising pull requests.
```
Comment (yhuiYH) [inline CONTRIBUTING.md]:
```suggestion
The GitHub "Issues" tab may also be used to discuss ideas surrounding particular features or changes, before raising pull requests.
```

**Pull request merged**